go-controller/pkg/cni/cni.go

```diff
@@ -149,11 +172,19 @@ func (pr *PodRequest) cmdCheck(podLister corev1listers.PodLister, useOVSExternal
 	if pr.IsSmartNIC {
 		annotCondFn = isSmartNICReady
 	}
-	annotations, err := GetPodAnnotations(pr.ctx, podLister, kclient, pr.PodNamespace, pr.PodName, annotCondFn)
+	rawAnnotations, err := GetPodAnnotations(pr.ctx, podLister, kclient, pr.PodNamespace, pr.PodName, annotCondFn)
+	if err != nil {
+		return nil, err
+	}
+	annotations, err := util.UnmarshalPodAnnotation(rawAnnotations)
 	if err != nil {
 		return nil, err
 	}
 
+	if _, err := setMACUint64(annotations.MAC.String(), &pr.InitialMAC); err != nil {
```

Comment:
why would we update the mac in Check?

Reply:
It uses the same logic as ADD and it also waits for pod flows, so we want to cancel the check for the same reasons we want ot cancel an outdated ADD.